src/noCommentedOutCodeRule.ts

```diff
@@ -42,51 +42,100 @@ class NoCommentedOutCodeRuleWalker extends ErrorTolerantWalker {
     }
 
     private scanCommentForCode(startPosition: number, commentText: string) {
-        if (this.textIsCode(this.cleanComment(commentText))) {
-            this.foundSuspiciousComment(startPosition, commentText);
+        const trimmedCommentText = this.trimTextLines(commentText);
+
+        if (this.isTextCode(trimmedCommentText)) {
+            this.handleSuspiciousComment(startPosition, commentText);
         }
     }
 
-    private textIsCode(text: string): boolean {
-        text = text.trim();
-        if (this.textIsSingleWord(text)) {
+    /**
+     * Removes spaces and comment delimiters at beginning
+     * and end of each line
+     */
+    private trimTextLines(text: string): string {
+        const lines = text.split('\n');
+
+        const trimAtStart = /^\s*\/*\**\s*/;
+        const trimAtEnd = /\s*\**\/*\s*$/;
+
+        const trimmedLines = lines.map(line => {
+            return line.replace(trimAtStart, '').replace(trimAtEnd, '');
+        });
+
+        return trimmedLines.join('\n');
+    }
+
+    private isTextCode(text: string): boolean {
+        if (this.isTextSingleWord(text)) {
             return false;
         }
-        if (this.textIsTslintFlag(text)) {
+        if (this.isTextTsLintFlag(text)) {
             return false;
         }
-        if (this.textIsNote(text)) {
+        if (this.isTextToDoLikeNote(text)) {
             return false;
         }
-        const sourceFile = ts.createSourceFile('', text, ts.ScriptTarget.ES5, true);
-        if (sourceFile) {
-            const { statements } = sourceFile;
-            const diagnostics: any[] = (<any>sourceFile).parseDiagnostics;
-            return statements.length > 0 && diagnostics.length === 0;
-        }
-        return false;
+
+        return this.isTextCodeWithoutErrors(text);
     }
 
-    private textIsSingleWord(text: string): boolean {
+    private isTextSingleWord(text: string): boolean {
```

Comment:
👍Good change.

I'd like to make a TSLint rule which sees a method's return type is `boolean` and checks the starting of the method name for `is` and other prefixes 🤣.